packages/docusaurus/src/client/exports/BrowserOnly.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useEffect, useState} from 'react';
+
+const BrowserOnly = ({children}) => {
+  const [isClient, setIsClient] = useState(false);
```

Comment:
What do you think about renaming it to `isMounted`?

Btw I'm wondering if we can use the `ExecutionEnvironment.canUseDOM` instead, then there's no need for a state right?

Reply:
Why so? This state is needed.

Reply:
What's the diff between doing your way and this:

```jsx
function BrowserOnly({children}) => {
  return ExecutionEnvironment.canUseDOM && children;
}
```

Reply:
It seemed to me that all the work of this component is to use `useEffect`, isn't it?

Reply:
Yes and I think we don't even need `useEffect` here. Can you try what I've proposed and see whether there's any difference? :/

Reply:
It seems that **none** of the options actually works, it really puzzled me and in complete dismay :face_with_thermometer: 

P.S. For the test, I used the `@u-wave/react-vimeo` component in the MDX file.

Reply:
So even your current implementation doesn't work? Luckily we test out first lol. Maybe the component is running some side effects that relies on browser environments being present during `import`. In that case there's no way around it.

Reply:
Not exactly, rather why something does not work, this component on some dependencies.

Reply:
After some investigation I found out that build fails if a component using a `window` object is **imported**, how is this possible? Therefore the `BrowserOnly` component is useless here :man_shrugging: